### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by DuchessInkeri**
In this MR: 

1. Added codestyle fixed;
2. Added readme.md;
3. Added .jar for: codestyle check (with xml), junit tests;
4. Some tests error edits;
5. Variable names corrected

### CODE REVIEW

**Comment by olesya-wo** [comment]
You should always provide a description for PR, especially if it didn't link with any particular issue

File: src/Main.java [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import Calculator.*;
-
 import java.util.Scanner;
 
-import static java.lang.StringTemplate.STR;
 
 public static void main() {
```
Comment (olesya-wo) [inline src/Main.java]: Don't make implicit classes, declare this class explicitly

File: src/Main.java [hunk 2]
```diff
@@ -21,17 +19,16 @@ public static void main() {
         var result = RPN.convert(tokens);
         System.out.println(STR."RPN:\n\{result}");
```
Comment (olesya-wo) [inline src/Main.java]: Don't use string templates here

File: src/Main.java [hunk 3]
```diff
@@ -21,17 +19,16 @@ public static void main() {
...
         var calculate = Calculate.calculate(tokens);
-        System.out.println(STR."Result: \{calculate.pop().doubleValue()}");
+        System.out.println(STR."Result: \{calculate.pop()}");
```
Comment (olesya-wo) [inline src/Main.java]: Method `calculate` should return a double, not an array

File: src/Calculator/Token.java [hunk 4]
```diff
@@ -43,8 +43,7 @@ public String toString() {
     public boolean equals(Object other) {
```
Comment (olesya-wo) [inline src/Calculator/Token.java:48]: Where did you use this?

File: src/Main.java [hunk 5]
```diff
@@ -21,17 +19,16 @@ public static void main() {
...
 
         boolean cont = true;
         while (cont) {
```
Comment (olesya-wo) [inline src/Main.java]: Please remove inner loop, it's unnecessary

**Comment by olesya-wo** [comment]
Please provide in your [README.md](https://github.com/Froidnote/RPN-Calculator/blob/master/README.md) this information:
- How to compile your app
- How to run it
- How to run all tests

And your algorithm is not quite RPN, it's [Shunting yard algorithm](https://en.wikipedia.org/wiki/Shunting_yard_algorithm).
Classic RPN doesn't support unary operators.

File: src/Calculator/Token.java (hunk 4 above)
Comment (DuchessInkeri) [reply src/Calculator/Token.java:48]: Used for ShuntingYardTest 
80: Assertions.assertEquals

File: src/Main.java (hunk 5 above)
Comment (Froidnote) [reply src/Main.java]: Исправлено в 4cad60ec4429809c87b56b2e684cab235a859922

File: src/Main.java (hunk 3 above)
Comment (Froidnote) [reply src/Main.java]: Исправлено в 4cad60ec4429809c87b56b2e684cab235a859922

File: src/Main.java (hunk 2 above)
Comment (Froidnote) [reply src/Main.java]: Исправлено в 4cad60ec4429809c87b56b2e684cab235a859922

File: src/Main.java (hunk 1 above)
Comment (Froidnote) [reply src/Main.java]: Corrected 4cad60ec4429809c87b56b2e684cab235a859922

**Comment by DuchessInkeri** [comment]
> * How to run all tests

In progress

**Comment by olesya-wo** [comment]
Don't use method `equals` on primitive types, use the regular `==`

File: src/Calculator/Calculate.java [hunk 6]
```diff
@@ -41,6 +44,10 @@ public static Stack<Double> calculate (ArrayList<Token> tokens) {
                         case "/":
                             answer.push(opR / opL);
                             break;
+
+                        default:
+                            throw new Error("Unrecognized operator%s".formatted(token.value));
```
Comment (olesya-wo) [inline src/Calculator/Calculate.java]:
```suggestion
                            throw new Error("Unrecognized operator '%s'".formatted(token.value));
```

Comment (olesya-wo) [reply src/Calculator/Calculate.java]: And also in all similar messages

File: src/Calculator/ShuntingYard.java [hunk 7]
```diff
@@ -1,35 +1,37 @@
...
+            } else if (token.type == Type.OPERATOR) {
                 if (i == 0 && token.value.equals("-")) {
                     token.value = "m";
                 } else if (i == 0 && token.value.equals("+")) {
                     token.value = "p";
                 } else if (!stack.empty()) {
                     var lastOP = stack.peek();
-                    if (!AfterParentheses) {
+                    if (!afterParentheses) {
                         if ((lastOP.type == Type.OPERATOR || lastOP.value.equals("(")) && token.value.equals("-") && !lastOP.value.equals("m") && !lastOP.value.equals("p") && !lastToken.isNumber()) {
```
Comment (olesya-wo) [inline src/Calculator/ShuntingYard.java]: This line is too long, check your linter settings, please

File: src/Calculator/ShuntingYard.java [hunk 8]
```diff
@@ -1,35 +1,37 @@
...
             case "*", "/" -> 2;
             case "+", "-" -> 1;
             default -> 0;
         };
     }
 
-    public static TokenCollection convert(ArrayList<Token>tokens) {
-        TokenCollection RPN = new TokenCollection();
+    public static TokenCollection convert(ArrayList<Token> tokens) {
+        TokenCollection rpn = new TokenCollection();
```
Comment (olesya-wo) [inline src/Calculator/ShuntingYard.java]: NIT: Something like `result` would be more appropriate, imo

File: src/Calculator/Token.java [hunk 9]
```diff
@@ -32,19 +32,22 @@ boolean isUnary() {
...
-        if (other == this) return true;
-        if (!(other instanceof Token)) return false;
-        Token otherToken = (Token) other;
+        if (other == null) {
+            return false;
+        }
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof Token otherToken)) {
```
Comment (olesya-wo) [inline src/Calculator/Token.java]: What does this line mean?

**Comment by olesya-wo** [comment]
Don't use `buffer = new StringBuilder()` when you want to reset `buffer` - this is bad for performance
Just set its length to zero with `buffer.setLength(0)`

File: src/Calculator/Type.java [hunk 10]
```diff
@@ -1,4 +1,4 @@
-package Calculator;
+package calculator;
```
Comment (olesya-wo) [inline src/Calculator/Type.java]: Please consider not making this a separate class
Make it part of Token and use as `Token.Type` if needed

File: src/Calculator/ShuntingYard.java [hunk 11]
```diff
@@ -14,34 +14,39 @@ private static int position(Token token) {
...
                             token.value = "p";
                         }
                     }
-                    while (lastOP.type == Type.OPERATOR) {
-                        if (token.isLeftAssociativity() && position(token) <= position(lastOP) || !token.isLeftAssociativity() && position(token) < position(lastOP)) {
-                            rpn.add(stack.pop());
+                    while (lastOP.type == Token.Type.OPERATOR) {
+                        if (token.isLeftAssociativity() && position(token) <= position(lastOP)
+                                || !token.isLeftAssociativity() && position(token) < position(lastOP)) {
+                            result.add(stack.pop());
```
Comment (olesya-wo) [inline src/Calculator/ShuntingYard.java]:
```suggestion
                        if (
                            token.isLeftAssociativity() && position(token) <= position(lastOP)
                            || !token.isLeftAssociativity() && position(token) < position(lastOP)
                        ) {
                            result.add(stack.pop());
```

File: src/Calculator/Token.java (hunk 9 above)
Comment (olesya-wo) [reply src/Calculator/Token.java]: Okay, I figured out, but this method still needs to be fixed

Comment (DuchessInkeri) [reply src/Calculator/Token.java]: Corrected

File: src/Main.java (hunk 3 above)
Comment (DuchessInkeri) [reply src/Main.java]: Corrected

File: src/Calculator/Calculate.java [hunk 12]
```diff
@@ -4,45 +4,46 @@
 import java.util.Stack;
 
 public class Calculate {
-    public static Stack<Double> calculate(ArrayList<Token> tokens) {
+    public static double calculate(ArrayList<Token> tokens) {
         TokenCollection result = ShuntingYard.convert(tokens);
-        Stack<Double> answer = new Stack<>();
+        Stack<Double> stack = new Stack<>();
+        double answer;
```
Comment (olesya-wo) [inline src/Calculator/Calculate.java]: Why did you declare it here?
You don't need this var anyway

File: src/Calculator/Calculate.java [hunk 13]
```diff
@@ -52,6 +53,7 @@ public static Stack<Double> calculate(ArrayList<Token> tokens) {
                 }
             }
         }
+        answer = stack.pop();
         return answer;
```
Comment (olesya-wo) [inline src/Calculator/Calculate.java]: Just `return stack.pop();`

File: .gitignore [hunk 14]
```diff
@@ -1,6 +1,7 @@
 ### IntelliJ IDEA ###
 out/
```
Comment (olesya-wo) [inline .gitignore:2]: Look, you've included this folder, but never used it
And your binary files after compilation now placed beside their .java files
You get mess
Please separate your source code (`src`) and your binary distribution files (`out`) during your compilation process

And after this, you won't need `*.class` filter here, only entire `out`
Eventually, your gitignore will look like this:
```
out/
*.iml
.idea/
.vscode/
```

File: README.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+To use program:
+
+* Change Directory to \src
+* Compile: ```javac Main.java```
+* Run: ```java Main.java```
+
+***
+To use tests:
+
+* Change Directory to \src
```
Comment (olesya-wo) [inline README.md]: Just provide as command
`cd src`

File: README.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+* Compile:
+  ```javac -cp junit-platform-console-standalone-1.9.3.jar;. test\CalculateTest.java; test\ShuntingYardTest.java```
+* Run: ```java -jar junit-platform-console-standalone-1.9.3.jar --class-path .;test --scan-class-path```
```
Comment (olesya-wo) [inline README.md]: Also add a command (and .jar) for checkstyle

File: README.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+
+* Change Directory to \src
+* Compile: ```javac Main.java```
+* Run: ```java Main.java```
+
+***
+To use tests:
+
+* Change Directory to \src
+* Compile:
```
Comment (olesya-wo) [inline README.md]: Write all commands consistently

`description`: `command`
or
`description`:
`command`

Don't mix both

File: src/Main.java [hunk 18]
```diff
@@ -1,40 +1,32 @@
...
+            if (string.equalsIgnoreCase("y")) {
                 System.out.println("Please enter your expression:");
-                cont = false;
-            } else if (answer.toLowerCase().equals("n")) {
-                cont = false;
+            } else if (string.equalsIgnoreCase("n")) {
                 return;
+            } else {
+                System.out.println("Invalid input");
+                break;
```
Comment (olesya-wo) [inline src/Main.java:28]: You made the exit from this loop with `return` above, but with `break` here
Why?

File: README.md [hunk 19]
```diff
@@ -6,14 +6,21 @@
...
-***
 To use tests:
 
-* Change Directory to \src
-* Compile:
-  ```javac -cp junit-platform-console-standalone-1.9.3.jar;. test\CalculateTest.java; test\ShuntingYardTest.java```
-* Run: ```java -jar junit-platform-console-standalone-1.9.3.jar --class-path .;test --scan-class-path```
+* cd src
+* Compile:```javac -d out -cp "junit-platform-console-standalone-1.9.3.jar;." test\CalculateTest.java test\ShuntingYardTest.java```
+* Run: ```java -jar junit-platform-console-standalone-1.9.3.jar --class-path "out" --scan-class-path```
```
Comment (olesya-wo) [inline README.md]: You don't need to brace "out" in quotes

File: README.md [hunk 20]
```diff
@@ -6,14 +6,21 @@
...
+
+---
+
+To use checkstyle:
+
+* cd RPN-Calculator
```
Comment (olesya-wo) [inline README.md]: This is unnecessary

File: checkstyle.xml [hunk 21]
```diff
@@ -0,0 +1,119 @@
...
+    <!-- Checks for Size Violations.                    -->
+    <!-- See http://checkstyle.sf.net/config_sizes.html -->
+    <module name="FileLength"/>
+
+    <!-- Checks for Headers                                -->
+    <!-- See http://checkstyle.sf.net/config_header.html   -->
+    <!-- <module name="Header"> -->
+    <!--   <property name="headerFile" value="${checkstyle.header.file}"/> -->
+    <!--   <property name="fileExtensions" value="java"/> -->
+    <!-- </module> -->
```
Comment (olesya-wo) [inline checkstyle.xml]: Always just remove commented code

File: .gitignore (hunk 14 above)
Comment (olesya-wo) [reply .gitignore:2]: I said how your file should look like

**Comment by olesya-wo** [comment]
> You should always provide a description for PR, especially if it didn't link with any particular issue

You still have to

File: README.md [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+* Run: java -jar checkstyle-10.16.0.jar -c checkstyle.xml src
```
Comment (olesya-wo) [inline README.md]: No code highlight for command

File: README.md [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+* cd src
+* Compile:```javac -d out Main.java```
+* Run: ```java -cp ./out Main.java```
+
+---
+
+To use tests:
+
+* cd src
+* Compile:```javac -d out -cp "junit-platform-console-standalone-1.9.3.jar;." test\CalculateTest.java test\ShuntingYardTest.java```
```
Comment (olesya-wo) [inline README.md]: Try to use a mask for source files, in order to get rid of enumerating each file separately
As shown here - https://www.baeldung.com/java-compile-multiple-files

File: README.md [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+
+### based on Shunting yard algorithm
+
+#### SDK: OpenJDK 21.0.2
+
+To use program:
+
+* cd src
+* Compile:```javac -d out Main.java```
+* Run: ```java -cp ./out Main.java```
```
Comment (olesya-wo) [inline README.md]: Not `Main.java`

File: README.md [hunk 25]
```diff
@@ -0,0 +1,26 @@
+## Command Line Interface Calculator
+
+### based on Shunting yard algorithm
+
+#### SDK: OpenJDK 21.0.2
+
+To use program:
```
Comment (olesya-wo) [inline README.md:7]: And also provide the step to pack your compiled `.class` files into `.jar` file with appropriate manifest that contains entry point
Then you can run this .jar file directly

**Comment by DuchessInkeri** [comment]
> Please provide in your [README.md](https://github.com/Froidnote/RPN-Calculator/blob/master/README.md) this information:
> 
> * How to compile your app
> * How to run it
> * How to run all tests
> 
> And your algorithm is not quite RPN, it's [Shunting yard algorithm](https://en.wikipedia.org/wiki/Shunting_yard_algorithm). Classic RPN doesn't support unary operators.

Done

**Review by olesya-wo — APPROVED** [review]

**Pull request merged**